src/clean-core/assert.hh

```diff
@@ -64,8 +70,9 @@
 #endif
 
 #ifdef CC_ENABLE_ASSERTIONS
-#define CC_UNREACHABLE(msg) \
-    (::cc::detail::assertion_failed({"unreachable code reached: " msg, CC_PRETTY_FUNC, __FILE__, nullptr, __LINE__}), CC_BREAK_AND_ABORT(), CC_BUILTIN_UNREACHABLE)
+#define CC_UNREACHABLE(msg)                                                                                                             \
+    (CC_DEBUG_BREAK(), ::cc::detail::assertion_failed({"unreachable code reached: " msg, CC_PRETTY_FUNC, __FILE__, nullptr, __LINE__}), \
+     ::cc::detail::perform_abort(), CC_BUILTIN_UNREACHABLE)
```

Comment:
Note: Flip order of debug break / assertion failed as in CC_DETAIL_EXECUTE_ASSERT